### ISSUE TITLE
Support generating typedefs as #[repr(transparent)]

### ISSUE THREAD

**Issue by TimNN** [rust-lang/rust-bindgen#1488]
macOS has a typedef like this:

```c
typedef int OSStatus;
```

I would prefer if the generated code looked something like this:

```rust
#[repr(transparent)]
pub struct OSStatus(c_int);
```

rather than

```rust
pub type OSStatus = c_int;
```

**Comment by emilio** [comment]
That sounds like a neat feature, I'd be happy to mentor it or what not :)

**Comment by pvdrz** [comment]
Closed via: https://github.com/rust-lang/rust-bindgen/issues/1488

**Comment by kulp** [comment]
> Closed via: #1488

I think @pvdrz meant to say #1658. Anyway I agree this is resolved.
